core/src/main/scala/org/apache/spark/api/r/RBackendHandler.scala

```diff
@@ -284,7 +286,7 @@ private[r] object JVMObjectTracker {
     objId
   }
 
-  def remove(id: String): Option[Object] = {
+  def remove(id: String) {
```

Comment:
Although it's normal to return the object that was removed from map methods, if this isn't desired, I'd declare this as `: Unit` to be clear